src/test/java/com/uid2/admin/store/writer/EncryptedScopedStoreWriterTest.java

```diff
@@ -0,0 +1,385 @@
+
+    @Mock
+    private Clock clock;
+
+    @Mock
+    private VersionGenerator versionGenerator;
+
+    @Mock
+    private IMetadataVersionedStore provider;
+
+    @Mock
+    private RotatingS3KeyProvider s3KeyProvider;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this);
+        cloudStorage = new InMemoryStorageMock();
+        FileStorageMock fileStorage = new FileStorageMock(cloudStorage);
+        fileManager = new FileManager(cloudStorage, fileStorage);
+        globalStore = mock(RotatingSiteStore.class);
+        jsonWriter = JsonUtil.createJsonWriter();
+
+        // Generate a valid 32-byte AES key
+        byte[] keyBytes = Random.getRandomKeyBytes();
+        String base64Key = Base64.getEncoder().encodeToString(keyBytes);
+        encryptionKey = new S3Key(1, testSiteId, 0, 0, base64Key);
+
+        Map<Integer, S3Key> mockS3Keys = new HashMap<>();
+        mockS3Keys.put(testSiteId, encryptionKey);
+
+        // Mock the reader
+        when(reader.getSnapshot()).thenReturn(mockS3Keys);
+
+        // Mock the s3KeyProvider
+        when(s3KeyProvider.getAll()).thenReturn(mockS3Keys);
+
+        // Mock the globalStore
+        when(globalStore.getAllSites()).thenReturn(new ArrayList<>());
+        when(globalStore.getMetadata()).thenReturn(new JsonObject());
+
+        // Initialize EncryptedScopedStoreWriter with the s3KeyProvider
+        encryptedScopedStoreWriter = new EncryptedScopedStoreWriter(
+                provider,
+                fileManager,
+                versionGenerator,
+                clock,
+                scope,
+                dataFile,
+                dataType,
+                s3KeyProvider
+        );
+    }
+
+    @Test
+    void testEncryptData() throws Exception {
+        String data = "test data";
+        byte[] secret = Base64.getDecoder().decode(encryptionKey.getSecret());
+        byte[] encryptedPayload = AesGcm.encrypt(data.getBytes(StandardCharsets.UTF_8), secret);
+
+        JsonObject encryptedJson = new JsonObject()
+                .put("key_id", encryptionKey.getId())
+                .put("encryption_version", "1.0")
+                .put("encrypted_payload", Base64.getEncoder().encodeToString(encryptedPayload));
+
+        assertEquals(encryptionKey.getId(), encryptedJson.getInteger("key_id"));
+        assertEquals("1.0", encryptedJson.getString("encryption_version"));
+        assertFalse(encryptedJson.getString("encrypted_payload").isEmpty());
+    }
+
+    @Test
+    void testUploadLogic() throws Exception {
+        Collection<Site> sites = oneSite;
+        String data = jsonWriter.writeValueAsString(sites);
+        JsonObject extraMeta = new JsonObject();
+
+        // Manually execute the logic from upload method
+        if (isEncrypted(data)) {
+            // If it's encrypted, simulate calling the superclass upload method
+            System.out.println("Calling super.upload with already encrypted data");
```

Comment:
Is this a test failure? Printing to system out doesn't tell me if the tests passed or failed.